src/test/operators/jit_operator/specialization/jit_repository_test.cpp

```diff
@@ -0,0 +1,69 @@
+#include <llvm/IRReader/IRReader.h>
+#include <llvm/Support/SourceMgr.h>
+
+#include "../../../base_test.hpp"
+#include "load_module.hpp"
+#include "operators/jit_operator/specialization/jit_repository.hpp"
+
+namespace opossum {
+
+class JitRepositoryTest : public BaseTest {
+ protected:
+  void SetUp() override {
+    _context = std::make_shared<llvm::LLVMContext>();
+    _module = load_module("src/test/llvm/virtual_methods.ll", *_context);
+  }
+
+  std::shared_ptr<llvm::LLVMContext> _context;
+  std::unique_ptr<llvm::Module> _module;
+};
+
+TEST_F(JitRepositoryTest, ProvidesAccessToDefinedFunctions) {
+  auto repository = JitRepository(std::move(_module), _context);
+
+  // Check that all defined methods in the class hierarchy are present in the bitcode repository.
+  // Virtual methods that are not implemented should cause a nullptr.
+  // See "src/test/llvm/virtual_methods.cpp" for the class hierarchy.
+  ASSERT_EQ(repository.get_function("_ZN4Base3fooEv"), nullptr);
```

Comment:
Is there a good way to have actual signatures instead of mangled names here?

Reply:
short answer: no
long answer: I am/was pretty annoyed by this as well and spend several days finding a solution for this a while back.
C++ name mangling is not part of LLVM itself but `libclang`. This would add another big dependency to Hyrise. And even withing clang the name mangling is not well isolated (there is no simple `get_mangled_name()` function, since the mangling relies on too much type information are requires a full compilation unit to be passed to `libclang`).
The solutions on stackoverflow either do implement their own name mangling with dodgy regular expressions or rely on generating code, calling `clang` and parsing the resulting bitcode string.